### PR TITLE
DMC 7 cards

### PR DESCRIPTION
**Pull request by Simisays**
The Ever-Changing 'Dane
Jedit Ojanen, Mercenary
Sivitri, Dragon Master
Dihada, Binder of Wills
General Marhault Elsdragon
Hazezon, Shaper of Sand
Rohgahh, Kher Keep Overlord

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/general_marhault_elsdragon.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Name:General Marhault Elsdragón
```
Comment (paulsnoops) [inline forge-gui/res/cardsfolder/upcoming/general_marhault_elsdragon.txt:1]: General Marhault Elsdragon

File: forge-gui/res/cardsfolder/upcoming/hazezon_shaper_of_sand.txt [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+ManaCost:R G W
+Types:Legendary Creature Human Warrior
+PT:3/3
+K:Desertwalk
+S:Mode$ Continuous | Affected$ Desert.YouCtrl | MayPlay$ True | AffectedZone$ Graveyard | Description$ You may play Desert lands from your graveyard
+T:Mode$ ChangesZone | ValidCard$ Desert | Origin$ Any | Destination$ Battlefield | Execute$ TrigToken | TriggerDescription$ Whenever a Desert enters the battlefield under your control, create two 1/1 red, green, and white Sand Warrior creature tokens.
+SVar:TrigToken:DB$ Token | TokenAmount$ 2 | TokenScript$ rgw_1_1_sand_warrior | TokenOwner$ You
+DeckHas:Ability$Token
+DeckHints:Type$Desert
+Oracle:Desertwalk (This creature can’t be blocked as long as defending player controls a Desert.)/nYou may play Desert lands from your graveyard./nWhenever a Desert enters the battlefield under your control, create two 1/1 red, green, and white Sand Warrior creature tokens.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/hazezon_shaper_of_sand.txt:11]: Wrong linebreaks

File: forge-gui/res/cardsfolder/upcoming/jedit_ojanen_mercenary.txt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+Name:Jedit Ojanen, Mercenary
+ManaCost:1 W U
+Types:Legendary Creature Cat Mercenary
+PT:3/3
+T:Mode$ ChangesZone | ValidCard$ Creature.Legendary+YouCtrl | Origin$ Any | Destination$ Battlefield | TriggerZones$ Battlefield | Execute$ TrigToken | OptionalDecider$ You | TriggerDescription$ Whenever CARDNAME or another legendary creature enters the battlefield, you may pay {G}. If you do, create a 2/2 green Cat Warrior creature token with forestwalk.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/jedit_ojanen_mercenary.txt:5]: ValidCard$ should include Card.Self, if for some reason it doesn't ETB as creature

File: forge-gui/res/cardsfolder/upcoming/dihada_binder_of_wills.txt [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+A:AB$ Dig | Cost$ SubCounter<3/LOYALTY> | Planeswalker$ True | DigNum$ 4 | Reveal$ True | AnyNumber$ True | Defined$ You | ChangeValid$ Card.Legendary | DestinationZone2$ Graveyard | SubAbility$ DBTreasure | RememberRevealed$ True | SpellDescription$ Reveal the top four cards of your library. Put any number of legendary cards from among them into your hand and the rest into your graveyard. Create a Treasure token for each card put into your graveyard this way.
+SVar:DBTreasure: DB$ Token | TokenAmount$ X | TokenScript$ c_a_treasure_sac | TokenOwner$ You | SubAbility$ DBCleanup
+A:AB$ GainControl | Cost$ SubCounter<11/LOYALTY> | Planeswalker$ True | AllValid$ Permanent.nonLand | AddKWs$ Haste | LoseControl$ EOT | Untap$ True | SpellDescription$ Gain control of all nonland permanents until end of. Untap them. They gain haste until end of turn.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/dihada_binder_of_wills.txt:9]: `until end of.` (also Oracle)
add `Ultimate$ True` please

File: forge-gui/res/cardsfolder/upcoming/dihada_binder_of_wills.txt [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+SVar:DBCleanup:DB$ Cleanup | ClearRemembered$ True
+SVar:X:Count$ValidGraveyard Card.IsRemembered
+DeckHas:Ability$Graveyard|Token & Type$Treasure
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/dihada_binder_of_wills.txt:12]:
```
DeckHas:Ability$Graveyard|Token|Sacrifice|LifeGain & Type$Treasure|Artifact
DeckHints:Type$Legendary
```

File: forge-gui/res/cardsfolder/upcoming/dihada_binder_of_wills.txt [hunk 6]
```diff
@@ -0,0 +1,13 @@
+Name:Dihada, Binder of Wills
+ManaCost:3 R W B
+Types:Legendary Planeswalker Dihada
+Loyalty:5
+Text:CARDNAME can be your commander.
+A:AB$ Pump | TargetMin$ 0 | TargetMax$ 1 | Cost$ AddCounter<2/LOYALTY> | KW$ Vigilance & Lifelink & Indestructible | Planeswalker$ True | ValidTgts$ Creature.Legendary | Duration$ UntilYourNextTurn | TgtPrompt$ Choose up to one legendary creature. | SpellDescription$ Up to one target legendary creature gains vigilance, lifelink, and indestructible until your next turn.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/dihada_binder_of_wills.txt:6]: TgtPrompt$ Select up to one target legendary creature

File: forge-gui/res/cardsfolder/upcoming/dihada_binder_of_wills.txt (hunk 5 above)
Comment (Northmoc) [reply forge-gui/res/cardsfolder/upcoming/dihada_binder_of_wills.txt:12]: Maybe DeckNeeds for the Legendary...

File: forge-gui/res/cardsfolder/upcoming/hazezon_shaper_of_sand.txt [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+ManaCost:R G W
+Types:Legendary Creature Human Warrior
+PT:3/3
+K:Desertwalk
+S:Mode$ Continuous | Affected$ Desert.YouCtrl | MayPlay$ True | AffectedZone$ Graveyard | Description$ You may play Desert lands from your graveyard
+T:Mode$ ChangesZone | ValidCard$ Desert | Origin$ Any | Destination$ Battlefield | Execute$ TrigToken | TriggerDescription$ Whenever a Desert enters the battlefield under your control, create two 1/1 red, green, and white Sand Warrior creature tokens.
+SVar:TrigToken:DB$ Token | TokenAmount$ 2 | TokenScript$ rgw_1_1_sand_warrior | TokenOwner$ You
+DeckHas:Ability$Token
+DeckHints:Type$Desert
+Oracle:Desertwalk (This creature can’t be blocked as long as defending player controls a Desert.\nYou may play Desert lands from your graveyard.\nWhenever a Desert enters the battlefield under your control, create two 1/1 red, green, and white Sand Warrior creature tokens.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/hazezon_shaper_of_sand.txt:11]: `’` > `'`

File: forge-gui/res/cardsfolder/upcoming/hazezon_shaper_of_sand.txt [hunk 8]
```diff
@@ -0,0 +1,11 @@
+Name:Hazezon, Shaper of Sand
+ManaCost:R G W
+Types:Legendary Creature Human Warrior
+PT:3/3
+K:Desertwalk
+S:Mode$ Continuous | Affected$ Desert.YouCtrl | MayPlay$ True | AffectedZone$ Graveyard | Description$ You may play Desert lands from your graveyard
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/hazezon_shaper_of_sand.txt:6]: YouOwn

File: forge-gui/res/cardsfolder/upcoming/hazezon_shaper_of_sand.txt [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+T:Mode$ ChangesZone | ValidCard$ Desert | Origin$ Any | Destination$ Battlefield | Execute$ TrigToken | TriggerDescription$ Whenever a Desert enters the battlefield under your control, create two 1/1 red, green, and white Sand Warrior creature tokens.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/hazezon_shaper_of_sand.txt:7]: Desert.YouCtrl

File: forge-gui/res/cardsfolder/upcoming/jedit_ojanen_mercenary.txt [hunk 10]
```diff
@@ -0,0 +1,8 @@
+Name:Jedit Ojanen, Mercenary
+ManaCost:1 W U
+Types:Legendary Creature Cat Mercenary
+PT:3/3
+T:Mode$ ChangesZone | ValidCard$ Card.Self,Creature.Legendary+YouCtrl| Origin$ Any | Destination$ Battlefield | TriggerZones$ Battlefield | Execute$ TrigToken | OptionalDecider$ You | TriggerDescription$ Whenever CARDNAME or another legendary creature enters the battlefield, you may pay {G}. If you do, create a 2/2 green Cat Warrior creature token with forestwalk.
+SVar:TrigToken:AB$ Token | Cost$ G | TokenAmount$ 1 | TokenOwner$ You | TokenScript$ g_2_2_cat_warrior_forestwalk
+Oracle:Whenever Jedit Ojanen, Mercenary or another legendary creature enters the battlefield, you may pay {G}. If you do, create a 2/2 green Cat Warrior creature token with forestwalk.
+DeckHas:Ability$Token
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/jedit_ojanen_mercenary.txt:8]: Move this line above Oracle, please

File: forge-gui/res/cardsfolder/upcoming/jedit_ojanen_mercenary.txt [hunk 11]
```diff
@@ -0,0 +1,8 @@
+Name:Jedit Ojanen, Mercenary
+ManaCost:1 W U
+Types:Legendary Creature Cat Mercenary
+PT:3/3
+T:Mode$ ChangesZone | ValidCard$ Card.Self,Creature.Legendary+YouCtrl| Origin$ Any | Destination$ Battlefield | TriggerZones$ Battlefield | Execute$ TrigToken | OptionalDecider$ You | TriggerDescription$ Whenever CARDNAME or another legendary creature enters the battlefield, you may pay {G}. If you do, create a 2/2 green Cat Warrior creature token with forestwalk.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/jedit_ojanen_mercenary.txt:5]: `Creature.Other+Legendary+YouCtrl`
No OptionalDecider, that is handled by cost in line 6

File: forge-gui/res/cardsfolder/upcoming/jedit_ojanen_mercenary.txt (hunk 10 above)
Comment (Northmoc) [reply forge-gui/res/cardsfolder/upcoming/jedit_ojanen_mercenary.txt:8]: `DeckHas:Ability$Token & Type$Warrior`
also `DeckHints:Type$Legendary`

File: forge-gui/res/cardsfolder/upcoming/jedit_ojanen_mercenary.txt (hunk 11 above)
Comment (Northmoc) [reply forge-gui/res/cardsfolder/upcoming/jedit_ojanen_mercenary.txt:5]: also fix spacing `YouCtrl|` > `YouCtrl |`

**Review by tool4ever — APPROVED** [review]

File: forge-gui/res/cardsfolder/upcoming/rohgahh_kher_keep_overlord..txt [hunk 12]
```diff
@@ -0,0 +1,12 @@
+Name:Rohgahh, Kher Keep Overlord
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/rohgahh_kher_keep_overlord..txt:1]: Double .. in filename

**Review by tool4ever — APPROVED** [review]

**Pull request merged**